examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,276 @@
+val_source = sorted(glob("data/source/*.jpg"))[-80:]
+val_targets = sorted(glob("data/target/*.jpg"))[-80:]
+
+height, width = 352, 608
+batch_size = 4
+
+
+def resize_with_pad(image, height, width):
+    return tf.image.resize_with_pad(
+        image=image, target_height=height, target_width=width
+    )
+
+
+def data_preprocess(source, target):
+    source_image = tf.io.decode_png(tf.io.read_file(source), channels=3)
+    target_image = tf.io.decode_png(tf.io.read_file(target), channels=3)
+    source_image = resize_with_pad(source_image, height=height, width=width)
+    target_image = resize_with_pad(target_image, height=height, width=width)
+    source_image = resnet_v2.preprocess_input(source_image)
+    target_image = resnet_v2.preprocess_input(target_image)
+    return source_image, target_image
+
+
+buffer_size = int(len(train_source) * 0.2)
+train_pipe = tf.data.Dataset.from_tensor_slices((train_source, train_targets))
+train_pipe = train_pipe.map(data_preprocess, tf.data.AUTOTUNE).cache()
+train_pipe = train_pipe.shuffle(buffer_size).batch(batch_size).repeat()
+
+
+val_pipe = tf.data.Dataset.from_tensor_slices((val_source, val_targets))
+val_pipe = val_pipe.map(data_preprocess, tf.data.AUTOTUNE).cache()
+val_pipe = val_pipe.batch(batch_size).repeat()
+
+
+train_step = len(train_source) // batch_size
+val_step = len(val_source) // batch_size
+
+fig, ax = plt.subplots(1, 2, figsize=(16, 8))
+ax = ax.flatten()
+for i, f in enumerate(val_pipe.take(1)):
+    noise, clean = f
+    noise = noise[i].numpy() + 1.0
+    noise = noise * 127.5
+    clean = clean[i].numpy() + 1.0
+    clean = clean * 127.5
+    ax[0].imshow(noise.astype(np.int))
+    ax[0].set_xlabel("source")
+    ax[1].imshow(clean.astype(np.int))
+    ax[1].set_xlabel("target")
+fig.text(x=0.45, y=0.78, s="Dataset Visualization", fontweight="bold")
+fig.set_tight_layout(None)
+
+"""
+## Model
+"""
+
+
+def conv_unit(x, filters=64, padding="same"):
+    x = keras.layers.Conv2D(filters=filters, kernel_size=3, padding=padding)(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.Activation("relu")(x)
+    return x
+
+
+def subpixel_upsample(x, target_channels, upscale_ratio):
+    estimated_filters = target_channels * (upscale_ratio ** 2)
+    x = keras.layers.Conv2D(estimated_filters, 3, padding="same")(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.LeakyReLU(0.2)(x)
+    x = tf.nn.depth_to_space(x, block_size=upscale_ratio)
+    return x
+
+
+def upscale_unit(previous_input, current_input, target_channels=3, upscale_ratio=2):
+    previous_low_scale = subpixel_upsample(
+        previous_input, target_channels=target_channels, upscale_ratio=upscale_ratio
+    )
+    current_high_scale = conv_unit(current_input, filters=target_channels)
+    concat_scales = tf.concat((previous_low_scale, current_high_scale), axis=-1)
+    feature_manifold = conv_unit(concat_scales, filters=target_channels)
+    return feature_manifold
+
+
+def denoiser():
```

Comment:
@anish9 The denoiser function doesn't take anything in the argument. So, I think you can simply state the function body statement outside.

Reply:
@innat I didn't grab that, can you elaborate on it further.

Reply:
@anish9 The statement of this function doesn't require to place within the function. For example, you can follow case 3 (stated below). 

```python
# case 1 
def my_func(x, y, z):
     outp = transform(x, y, z)
     return outp

# case 2 
def my_func():
     x, y, z = built-in-method()
     outp = transform(x, y, z)
     return outp

# case 3 
x, y, z = built-in-method()
outp = transform(x, y, z)
````

Reply:
@anish9 yes, but you can skip if you like.